troposphere/static/js/components/modals/instance/launch/components/ResourcesForm.jsx

```diff
@@ -135,13 +135,11 @@ const ResourcesForm = React.createClass({
         </label>
         );
     },
-    render: function() {
-        let { identity, identityList, onIdentityChange, providerSize, providerSizeList, onSizeChange, } = this.props;
-        return (
-        <form>
-            {globals.USE_ALLOCATION_SOURCES
-             ? this.renderAllocationSourceMenu()
-             : null}
+    renderIdentity() {
```

Comment:
Right, so when you say "sprinkle" you're referring to the fact that there would be redundancy, which is true. I think weak abstractions are worse than redundancy. A method that conditionally does nothing is a poor abstraction. Every method could use this functionality, but in reality the solution is just not to call it.